docs/contributing/index.rst

```diff
@@ -4,200 +4,464 @@
+
+    Note that not all issues which need contributors will have this tag.
+
+Making changes
+---------------
+
+The preferred way to contribute to geomstats is to fork the `main
+repository <https://github.com/geomstats/geomstats/>`__ on GitHub,
+then submit a "pull request" (PR).
+
+Follow the setup guidelines detailed in `getting the code`_. In this referenced section, 
+we explain how to locally install geomstats, and how to set up your git repository:
+The next steps now describe the process of modifying code and submitting a PR:
+
+#. Synchronize your master branch with the upstream master branch:
+
+    .. code-block:: bash 
 
         $ git checkout master
         $ git pull upstream master
 
-6. Create a feature branch to hold your development changes::
+#. | Create a feature branch to hold your development changes:
+
+    .. code-block:: bash 
 
         $ git checkout -b my_feature
 
    and start making changes. Always use a feature branch. It's good
    practice to never work on the ``master`` branch!
 
-7. Develop the feature on your feature branch on your computer, using Git to
+#. | Develop the feature on your feature branch on your computer, using Git to
    do the version control. When you're done editing, add changed files using
-   ``git add`` and then ``git commit``::
+   ``git add`` and then ``git commit``:
+
+    .. code-block:: bash 
 
        $ git add modified_files
        $ git commit -m "Add my feature"
 
    to record your changes in Git. Your commit message should respect the `good
    commit messages guidelines <https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project>`_.
-   Then push the changes to your GitHub account with::
+   Then push the changes to your GitHub account with:
+
+    .. code-block:: bash 
 
-       $ git push -u origin my-feature
+         $ git push -u origin my-feature
 
-8. Follow `these
-   <https://help.github.com/articles/creating-a-pull-request-from-a-fork>`_
+#. Follow `these <https://help.github.com/articles/creating-a-pull-request-from-a-fork>`_
    instructions to create a pull request from your fork. This will send an
    email to the committers. You may want to consider sending an email to the
    mailing list hi@geomstats.ai for more visibility.
 
 It is often helpful to keep your local feature branch synchronized with the
 latest changes of the main geomstats repository. If there are only a few new
-commits in the master branch, use::
+commits in the master branch, use:
+
+    .. code-block:: bash 
 
-    $ git fetch upstream
-    $ git rebase upstream/master
+      $ git fetch upstream
+      $ git rebase upstream/master
 
 Subsequently, you might need to solve potential conflicts.
 If it's been a while since you've last updated your branch, it might be easier
-to merge the master branch into yours::
+to merge the master branch into yours:
 
-    $ git fetch upstream
-    $ git merge upstream/master
+    .. code-block:: bash 
 
```

Comment:
we should mention pre-commit, black and flake8 here (related with point 5 below). To avoid repeating information we can just link people there (or write things here and send people back in that point). The way it is know it looks a bit hidden.